src/main/java/jenkins/branch/BranchNameContributor.java

```diff
@@ -46,9 +50,26 @@ public void buildEnvironmentFor(Job j, EnvVars envs, TaskListener listener) thro
         if (parent instanceof MultiBranchProject) {
             BranchProjectFactory projectFactory = ((MultiBranchProject) parent).getProjectFactory();
             if (projectFactory.isProject(j)) {
+                SCMHead head = projectFactory.getBranch(j).getHead();
                 // Note: not using Branch.name, since in the future that could be something different
                 // than SCMHead.name, which is what we really want here.
-                envs.put("BRANCH_NAME", projectFactory.getBranch(j).getHead().getName());
+                envs.put("BRANCH_NAME", head.getName());
+                ChangeRequestAction cr = head.getAction(ChangeRequestAction.class);
```

Comment:
Correct. Similar to the situation for `Actionable`s, where you can override `getActions`, etc., from the inside; or use `TransientActionFactory` from the outside. `Actionable` also exposes `addAction` from the outside, though this presumes that there is mutable persisted action storage, an imposition I did not really want to apply to `SCMHead` (besides the other API problems with `Actionable` mentioned in a linked JIRA ticket).